container/libcontainer/compatibility.go

```diff
@@ -159,8 +195,22 @@ func ReadConfig(dockerRoot, dockerRun, containerID string) (*configs.Config, err
 
 		var state libcontainer.State
 		err = json.Unmarshal(out, &state)
-		if err != nil {
-			return nil, err
+		if err != nil && strings.Contains(err.Error(), "json: cannot unmarshal") {
+			var oldstate oldState
```

Comment:
Can we take this out into a separate function? Better readability & easier to test.